frontend/lib/identification/qr_code_utils.dart

```diff
@@ -26,20 +28,41 @@ class QrCodeUtils {
   }
 
   List<int> _cardInfoToBinary(CardInfo cardInfo) {
-    final buffer = Uint8List(12).buffer;
-    final data = ByteData.view(buffer);
-    var offset = 0;
+    final json = messageToJsonObject(cardInfo);
+    final canonicalizedJsonString = JsonCanonicalizer().canonicalize(json);
+    return utf8.encode(canonicalizedJsonString);
+  }
 
-    final expirationDay = cardInfo.expirationDay;
-    data.setUint32(offset, expirationDay, Endian.little);
-    offset += 4;
-    data.setInt32(offset, cardInfo.extensions.extensionBavariaCardType.cardType.value, Endian.little);
-    offset += 4;
-    data.setInt32(offset, cardInfo.extensions.extensionRegion.regionId, Endian.little);
-    offset += 4;
+  Map<String, dynamic> messageToJsonObject(GeneratedMessage message) {
+    if (message.unknownFields.isNotEmpty) throw ArgumentError("Unknown field");
```

Comment:
Probably we should return a more distinct error here and catch it, and show something like "Your app is likely out of date.".